src/linter_options.h

```diff
@@ -0,0 +1,98 @@
+//
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#ifndef SRC_LINTER_OPTIONS_H_
+#define SRC_LINTER_OPTIONS_H_
+
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace zetasql::linter {
+
+class LinterOptions {
+ public:
+  // Returns if the linter check should be active
+  // in <position>.
+  bool IsActive(int position) const;
```

Comment:
Ahm, will you than create LinterOptions instance for each check separately? May be, it should be renamed to CheckOptions? Or having a nested structure:
```
class LinterOptions
  // enable-disable for individual checks
  map<CheckType, EnabledRange>
  // generic options for each check
  int tab_size;
  char line_delimiter;
  ...
```
?

Reply:
It was originally in nested structure. Then I thought passing options that only that check needs makes the code better. Now, with using "const LinterOptions &" type there is no harm to pass everything. I will make it nested again.